### PR TITLE
Feat ens avatar addedcomp

### PR DESCRIPTION
**Pull request by 3scava1i3r**
---
name: ''feat:Added ENSAvatar"
about: Issue #688 
---

### Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [x] My code is conform the [code style](https://github.com/web3ui/web3uikit/blob/master/CONTRIBUTE.md)
- [x] I have made corresponding changes to the documentation
- [x] I have updated Typescript definitions when needed

### Issue Description

This PR solves Issue #688 about adding an avatar for people that have ENS Avatar
Related issue: #688 

### Solution Description

I worked on adding the logic/component of ENSAvatar where we use blockies as a backup.If our consumer has ENSAvatar enabled the component would show the same otherwise the address is taken as seed for blockie

### CODE REVIEW

**Comment by AbhinavMV** [comment]
Hey @3scava1i3r
Cheers to your first PR. 🍻 
sorry but have some comments. 😅
Also please use prettier extension it will clean the code 😄

**Comment by 3scava1i3r** [comment]
Thank you @AbhinavMV for the feedback,did i miss something(comments)?
Added prettier to make the code clean

File: src/components/ENSAvatar/ENSAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    useEffect(() => {
+        if (account && chainId) {
+
+            web3?.lookupAddress(account).then((result) => {
+                setensName(result)
+
+            })
+
+            web3?.getAvatar(String(ensName)).then((r) => {
```
[the remark this answers is not in the document]
Comment (3scava1i3r) [reply src/components/ENSAvatar/ENSAvatar.tsx]: please check the changes made

File: src/components/ENSAvatar/ENSAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            })
+        }
+    }, [account, chainId]);
+
+
+
+    return (
+        <>
+        {(AvatarURI == String(null)) ? 
+        (<>
```
[the remark this answers is not in the document]
Comment (3scava1i3r) [reply src/components/ENSAvatar/ENSAvatar.tsx]: took that out

**Comment by BillyG83** [comment]
nice feedback @AbhinavMV if @3scava1i3r has addredd your concerns please approve and merge if you are happy

**Comment by AbhinavMV** [comment]
TY @BillyG83 , Looks good for v1.
I will update over this, it needs some styling props such as custom user width and height.
Also I was thinking that instead of showing avatar of current wallet address we can give address as input prop and it will fetch the avatar. This will make it reusable in other components

@3scava1i3r great work, if you want to do the above-mentioned points, let me know else I will merge this for now. 
Btw could you remove unused code? workflow is failing because of this.

File: src/components/ENSAvatar/ENSAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from 'react';
+import { ENSAvatarprops } from './types';
+import { useMoralis, useEnsAddress, useEnsName } from 'react-moralis';
+import Blockies from 'react-blockies';
```
Comment (AbhinavMV) [inline src/components/ENSAvatar/ENSAvatar.tsx:4]: useEnsName needs to be removed

File: src/components/ENSAvatar/ENSAvatar.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import { ENSAvatarStyles } from './ENSAvatar.styles';
+
+const { NonAvatarDiv, AvatarImg } = ENSAvatarStyles;
+
+const ENSAvatar: React.FC<ENSAvatarprops> = (props) => {
+    const { account, chainId, web3 } = useMoralis();
+    const { name, isLoading, error } = useEnsAddress(String(account));
+
```
Comment (AbhinavMV) [inline src/components/ENSAvatar/ENSAvatar.tsx:12]: isLoading and error needs to be removed

**Comment by 3scava1i3r** [comment]
As you mentioned @AbhinavMV i can work on a v2 where all the above mentioned props can be added.I'd like to work on these components.Removing all the console logs and the unnecessary variables.
How can i test this locally so that it isn't shown in the pr next time though?

**Comment by AbhinavMV** [comment]
> As you mentioned @AbhinavMV i can work on a v2 where all the above mentioned props can be added.I'd like to work on these components.Removing all the console logs and the unnecessary variables. How can i test this locally so that it isn't shown in the pr next time though?

Nice, you are awesome 🔥 

please run `yarn build` locally.

**Comment by AbhinavMV** [comment]
Nice work here @3scava1i3r

**Pull request merged**